astropy/units/quantity_helper/helpers.py

```diff
@@ -234,11 +234,25 @@ def helper_frexp(f, unit):
 
 
 def helper_multiplication(f, unit1, unit2):
-    return [None, None], _d(unit1) * _d(unit2)
+    match unit1, unit2:
+        case None, None:
+            return [None, None], dimensionless_unscaled
```

Comment:
This `case` is required for code like
```python
>>> from astropy import units as u
>>> import numpy as np
>>> q = [1, 2, 3] * u.one
>>> np.multiply(7, 8, out=q)
```
but apparently there are no tests for this.

Reply:
arrays in, quantity out is definitely tested for, but not for all ufuncs. 

It might be worth adding that parametrized for all the ufuncs in the future... (When trying to write new unit tests for Quantity2.0, I noticed that really what we have could use some care; I may try to port those new tests back.)